### DISCUSSION TITLE
How to remap NeoTree to <C-1> in insert mode?

### DISCUSSION THREAD

**Discussion by Gieted** [LazyVim/LazyVim#4718] [Q&A]
Hi, how can I remap "Explorer NeoTree (root dir)" to C-1 in insert mode?

```lua
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      { "<C-1>", "<leader>fe", desc = "Explorer NeoTree (root dir)", remap = true },
      { "<C-1>", "<leader>fe", desc = "Explorer NeoTree (root dir)", remap = true, mode = { "i" } },
    },
  },
}
```

The first mapping (for normal mode) works just fine, but the second doesn't.

**Comment by dotfrag** [comment]
You need to exit insert mode first.

```lua
    { "<C-1>", "<esc><leader>fe", desc = "Explorer NeoTree (root dir)", remap = true, mode = { "i" } },
```

**Reply by Gieted** [reply]
No, it doesn't work. It will go to normal mode, but NeoTree isn't opened. My complete config for this plugin is:
```
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      { "<C-1>", "<leader>fe", desc = "Explorer NeoTree (root dir)", remap = true },
      { "<C-1>", "<esc><leader>fe", desc = "Explorer NeoTree (root dir)", remap = true, mode = { "i" } },
    },
  },
}
```

**Reply by dotfrag** [reply]
You are not using the code I provided above. You need `<esc>` in there.

**Reply by Gieted** [reply]
Sorry, I copied the wrong thing, actually even with <esc> it's still not working

**Reply by Gieted** [reply]
Or actually it is working, but the weird thing is that you have to use the normal mode shortcut first after each neovim launch, otherwise it won't work. I'm wondering, can it be because of the lazyloading or something?

**Reply by dotfrag** [reply]
I'm on my phone, I'll check tomorrow and get back to you.

**Reply by dpetka2001** [reply]
It works fine for me

[Screencast_2024-11-09-00-57-09.webm](https://github.com/user-attachments/assets/2061fcc7-3e40-46fd-a096-6f6ead2721e2)

I changed the mapping to `<M-1>` because `<C-1>` is used by my terminal. No need to use the normal mapping first.

**Reply by Gieted** [reply]
The funny thing is that with `<M-1>` it's working just fine also for me, then the problem seems to be with just the `<C-1>` binding. Do you know why it might be the case?

**Reply by dpetka2001** [reply]
I temporarily disabled my terminal mapping and used it in Neovim for Neo-tree and it works as expected as well. What does `:verbose imap <C-1>` show? Maybe there's a conflict with some other mapping?

**Reply by Gieted** [reply]
![obraz](https://github.com/user-attachments/assets/c1de3d4b-580c-4009-9b45-f9735efa2732)

**Reply by dpetka2001** [reply]
What terminal are you using?

**Reply by Gieted** [reply]
I'm using Neovide client

**Reply by dpetka2001** [reply]
That might be the problem then. Can you try in a terminal with normal Neovim and see if the same behavior occurs? I'm not able to reproduce on my end with Neovim nightly on kitty terminal (I've also tried Wezterm and it also works there as expected).

**Reply by dotfrag** [reply]
This works for me both in terminal (kitty) and neovide client. I'm on Linux if that means anything.

**Reply by Gieted** [reply]
I cannot find a terminal for windows where ctrl+numbers would work. Also the strange thing is that sometimes it works as if <leader> was skipped, because it jumps to next letter "e", which is "<esc>fe", and sometimes it works as if <esc> was skipped as it just types " fe".

**Reply by Gieted** [reply]
Isn't this because terminal programs cannot distinguish between escape key press and escape sequence (like arrows), so they need to wait for some time for esc to actually work? This would make total sense then why this isn't working, although it's not clear to me why this works for you.

**Reply by dpetka2001** [reply]
No idea really. You should ask this in Neovide repo, because they are more knowledgeable about its inner workings.

**Reply by Gieted** [reply]
Oh man, thank you for help. I think I'll have to write my own editor then.

**Reply by dotfrag** [reply]
I know this is not helpful, but I've been on Windows many years, and most, if not all power user related stuff is garbage. Terminals, editors, cli tools.. Have you tried wezterm or alacritty? [Ghosty](https://github.com/ghostty-org) is also coming out soon.
